### ISSUE TITLE
Problem to create png/jpeg image in node.js with 3D plots

### ISSUE THREAD

**Issue by linev** [root-project/jsroot#257]
Up to now only `makeSVG` function was provided.

It is not possible to create images for different objects types.

**Comment by linev** [comment]
Now `makeImage` function will be introduced. It supports 'png', 'svg' and 'jpeg' image formats.

Usage examples are shown in [make_image.htm](https://github.com/root-project/jsroot/blob/master/demo/make_image.htm) and [make_image.js](https://github.com/root-project/jsroot/blob/master/demo/node/make_image.js) demos.

**Comment by linev** [comment]
Unfortunately, there is one unresolved issue with `node-canvas` component, used for image production in `node.js`.

It is reported now here: https://github.com/Automattic/node-canvas/issues/2254

Because of this limitation embed images like lego plots on the canvas are not working in `node.js`.
I hope solution will be find - until when issue will stay open.

**Comment by linev** [comment]
Function appears in https://github.com/root-project/jsroot/releases/tag/7.4.0

Keep open until issue with node-canvas is not resolved.

**Comment by OmarMesqq** [comment]
Hello, @linev I also faced this issue on older Linux distros (Ubuntu <= 20.04). I investigated the issue using Docker and left a comment at `node-canvas` issues:

https://github.com/Automattic/node-canvas/issues/1607#issuecomment-3568077756

Would you still like a manual fix for this or is upgrading the OS a plausible alternative?

**Comment by linev** [comment]
> In fact, node-canvas uses node-gyp to pull binaries during its build if I'm not mistaken.

If it so - can one include `tumbleweed` or similar platform to `node-canvas` builds?

**Comment by OmarMesqq** [comment]
> > In fact, node-canvas uses node-gyp to pull binaries during its build if I'm not mistaken.
> 
> If it so - can one include `tumbleweed` or similar platform to `node-canvas` builds?

Yes, it's possible. However that would mean submitting a PR to `node-canvas` so they update their CI scripts and generate binaries for newer Linux distros (with the fixed `librsvg`). This could take some time and get rejected, since they are probably linking against older libraries for maximum compatibility with older systems.

We can see the discrepancy regarding the broken library when building `node-canvas` (such as in my Dockerfile) and when simply cloning `jsroot` and running `npm install`.

1. Standard install:
```
ldd node_modules/canvas/build/Release/canvas.node
librsvg-2.so.2 => /app/jsroot/node_modules/canvas/build/Release/librsvg-2.so.2
```
it uses its own, baked-in broken, yet more compatible lib

2. Building from source:
```
ldd node_modules/canvas/build/Release/canvas.node
librsvg-2.so.2 => /lib64/librsvg-2.so.2
```
uses the system's (in your case, Tumbleweed's) updated and fixed library.

We could try swapping the SVG rendering engine from`node-canvas` in Linux to another library. I found [this one](https://github.com/thx/resvg-js). It ships its own dependencies, akin to a static library, so there would be no headaches regarding compatibility across different Linux versions.

What do you think? We can add a simple `if` check in `jsroot`'s `makeImage` that tries this new library and falls back to `node-canvas`, so we don't introduce any breaking change.

**Comment by linev** [comment]
Sorry for long delay - I was in vacations.

I will check if `resvg-js` works for all our use-cases. 
If it does that it promise - then it can be real solution for us.

**Comment by OmarMesqq** [comment]
Sure, I will retarget my fix branch to `dev`. Do you prefer me to keep the local variable `RESVG_FEATURE_FLAG` (feature flag for using the `resvgjs` backend), or should I completely eliminate the `node-canvas` branch?

Pseudocode:
```javascript
// Inside svgToImage function in BasePainter.mjs

const RESVG_FEATURE_FLAG = true;
      
if (RESVG_FEATURE_FLAG) {
     // resvgjs branch
} else {
    // node-canvas branch
}
```

I also reverted the whitespace/formatting diff changes that my editor was doing, thanks for pointing that out!
